src/pools/D3M4626TypePool.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-FileCopyrightText: © 2021 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPool.sol";
+import "forge-std/interfaces/IERC4626.sol";
+
+interface VatLike {
+    function live() external view returns (uint256);
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface D3mHubLike {
+    function vat() external view returns (address);
+    function end() external view returns (EndLike);
+}
+
+interface EndLike {
+    function Art(bytes32) external view returns (uint256);
+}
+
+contract D3M4626TypePool is ID3MPool {
```

Comment:
I know we are trying to avoid styling changes, but considering this file is gonna be audited and then we won't want to touch it I'm mentioning this now:
Could we try to have a more general homogeneous order of things compared to the other pools?
Also the `wards` storage slot uses to be the 0 in almost all Maker contracts.

Reply:
Like the ordering? @MathisGD please rearrange to match the function / storage ordering of the other contracts if there is extra time.

Reply:
Fixed to match Spark pool to make diff easier

Reply:
Looks good! Just a minor remaining thing, if we could switch onlyHub with auth. (don't hate me :))

Reply:
thought I did this. np, changed